mappings/net/minecraft/entity/mob/WaterCreatureEntity.mapping

```diff
@@ -1,8 +1,15 @@
 CLASS net/minecraft/class_1480 net/minecraft/entity/mob/WaterCreatureEntity
 	METHOD method_35174 canSpawnUnderground (Lnet/minecraft/class_1299;Lnet/minecraft/class_5425;Lnet/minecraft/class_3730;Lnet/minecraft/class_2338;Ljava/util/Random;)Z
 		ARG 0 entityType
+		ARG 1 world
 		ARG 2 spawnReason
 		ARG 3 pos
 		ARG 4 random
+	METHOD method_37359 hasNoLight (Lnet/minecraft/class_5425;Lnet/minecraft/class_2338;)Z
+		ARG 0 world
+		ARG 1 pos
+	METHOD method_37360 hasStoneFloor (Lnet/minecraft/class_2338;Lnet/minecraft/class_5425;)Z
```

Comment:
i heard that the spawn requirement isn't the stone floor, but a few layers of stones below the floor?

Reply:
I don't think the code reads so.

Spawning condition from the code:
1. for-loop 5 times:
2. If the block is BASE_STONE_OVERWORLD, return true.
3. If the block is NOT water, return false.
4. If the loop ends (without finding one), return false.

The loop ends when it finds stone underwater. (It can technically return true for 1 layer of stone without water, but SpawnRestriction has IN_WATER.) So I think this can be called "stone floor".

Reply:
Right, mojang says within 5 blocks below the spawning space

Reply:
This sounds good to put in an `@implNote`.